src/compositions/useFavicon.ts

```diff
@@ -1,9 +1,33 @@
 import { onUnmounted, watchEffect, Ref, ref } from 'vue'
 import { StateType } from '@/models/StateType'
 
+function getPreferredColorScheme(): 'dark' | 'light' | 'no-preference' {
+  if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+    return 'dark'
+  } if (window.matchMedia('(prefers-color-scheme: light)').matches) {
+    return 'light'
+  }
+  return 'no-preference'
+}
+
 export function useFavicon(stateType: StateType | null | Ref<StateType | null | undefined>): void {
-  const favicon16 = document.getElementById('favicon-16')
-  const favicon32 = document.getElementById('favicon-32')
+  let favicon16: HTMLElement | null
+  let favicon32: HTMLElement | null
+
+  switch (getPreferredColorScheme()) {
+    case 'dark':
+      favicon16 = document.getElementById('favicon-16-dark')
+      favicon32 = document.getElementById('favicon-32-dark')
+      break
+    case 'light':
+      favicon16 = document.getElementById('favicon-16-light')
+      favicon32 = document.getElementById('favicon-32-light')
+      break
+    default:
+      favicon16 = document.getElementById('favicon-16-no-preference')
+      favicon32 = document.getElementById('favicon-32-no-preference')
+      break
```

Comment:
```suggestion
```
Can remove these in favor of the case above

Reply:
Any objection to removing the `prefers-color-scheme:no-preference` icon in the prefect and nebula-ui repos to go alongside this?  It looks like it's deprecated anyways and not supported anywhere but safari (which seems to be doing its own thing with favicons and ignoring the useFavicon change anyways).

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

Safari in light and dark when all other browsers are updating the favicon:
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/40272060/235196967-3b46a2ff-2669-4408-b18b-443b0a08a844.png">

<img width="1048" alt="image" src="https://user-images.githubusercontent.com/40272060/235197235-94a300c3-b292-4184-9b02-204fbf51c2f5.png">

Reply:
No objection from me - Safari might handle favicons a little differently, we've got a `mask-icon` link in the head which might be what Safari is pulling instead but I'm not sure